src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -90,6 +90,17 @@ public SearchOptions Create(string compartmentType, string compartmentId, string
                     // Query parameter with empty value is not supported.
                     unsupportedSearchParameters.Add(query);
                 }
+                else if (query.Item1 == KnownQueryParameterNames.Total)
+                {
+                    if (string.Compare(query.Item2, TotalType.Accurate.ToString(), StringComparison.OrdinalIgnoreCase) == 0)
```

Comment:
Could we convert the `TotalType` to a normal enum and do something like `query.Item2.GetValueByEnumLiteral<TotalType>()` to get the enum value?

Reply:
Good call on getting rid of the weird TotalType class.

`GetValueByEnumLiteral` didn't agree with case sensitivity, but `Enum.TryParse` seems to do the job. Let me know what you think of e6364ac.